File: app/src/main/java/org/scp/gymlog/util/WeightFormatter.java
package org.scp.gymlog.util;

import lombok.Getter;

@Getter
public class WeightFormatter {
    private boolean forceScale;
    private int scale;
    private boolean exactScale;

    public static final WeightFormatter DEFAULT_WEIGHT_FORMATTER = new WeightFormatter(false);
    public static final WeightFormatter EXACT_FORMATTER = new WeightFormatter(true);
    public static final WeightFormatter TWO_DECS_FORMATTER = new WeightFormatter(2);

    private WeightFormatter(boolean exactScale) {
        this.exactScale = exactScale;
    }

    public WeightFormatter(int scale) {
        this.forceScale = true;
        this.scale = scale;
    }
}

File: app/src/main/java/org/oar/gymlog/util/WeightFormatter.kt
package org.oar.gymlog.util

class WeightFormatter {

    companion object {
        val DEFAULT_WEIGHT_FORMATTER = WeightFormatter(false)
        val EXACT_FORMATTER = WeightFormatter(true)
        val TWO_DECS_FORMATTER = WeightFormatter(2)
    }

    var forceScale = false
    var scale = 0
    var exactScale = false

    constructor(exactScale: Boolean) {
        this.exactScale = exactScale
    }

    constructor(scale: Int) {
        forceScale = true
        this.scale = scale
    }
}